mesonbuild/dependencies/base.py

```diff
@@ -2262,11 +2242,11 @@ def log_tried(self):
         return 'framework'
 
 
-def get_dep_identifier(name, kwargs, want_cross):
-    identifier = (name, want_cross)
+def get_dep_identifier(name, kwargs, for_machine: MachineChoice) -> Tuple:
```

Comment:
Tuple of what?

Reply:
Look below it's a dynamic heterogenous mess:
```python
for ...:
    identifier += (key, value)
```

Reply:
I'd be just as happy with `Tuple[Any, Any]` if we really don't know what the types are, at least we're being explicit about what we don't know :)

Reply:
Error I thought this was a tuple with unknown length. Am i misinterpreting what `+=` does here?